### ISSUE TITLE
ava/assertion-arguments argues about explicit call of t.end() in test.cb()

### ISSUE THREAD

**Issue by florianb** [avajs/eslint-plugin-ava#164]
First of all i am not sure if this is a bug report or a feature request. I ended up deactivating the `assertion-arguments`-rule.

Given the following condensed test-code example, where i want to test if the method takes path `b`:

```js
test.cb(t => {
  m({
    a: {
      callback: () => t.end(true)
    },
    b: {
      callback: () => t.end(false)
    }
  });
});
```

`ava/assertion-arguments` complains: Too many arguments. Expected at most 0.

I know the documentation somehow exposes `t.end()` as single callback handler without any parameters. But it also says if "any truthy value [will be] passed as the first argument to t.end" it treats it as error.

@jfmengels already mentioned using t.fail/t.success instead. Unfortunately this is prohibited by `ava/test-ended`: Callback test was not ended. Make sure to explicitly end the test with t.end().

It's probably just a decision which rule to adjust, however. If there's something i can do. Let me know.

**Comment by jfmengels** [comment]
> @jfmengels already mentioned using t.fail/t.success instead. Unfortunately this is prohibited by ava/test-ended: Callback test was not ended. Make sure to explicitly end the test with t.end().

I meant to do something like this instead.
```js
test.cb(t => {
  m({
    a: {
      callback: () => t.end()
    },
    b: {
      callback: () => t.fail()
    }
  });
});
```

> AVA will consider any truthy value passed as the first argument to t.end to be an error.

Indeed, this is simply something that I missed when I wrote the rule, influenced by the fact that I had never seen any args passed to it. I agree that this is a bug and that it should be fixed.

**Comment by florianb** [comment]
![](https://media.giphy.com/media/3o6wrzcyVdjrEC5UaY/giphy.gif)
